mantidproject/mantid
# Split Reflectometry interface into tabs

Description of work: This PR refactors the existing functionality into two different tabs. The `Runs` tab contains the `Search` list and `Process` table. The `Settings` tab has text boxes to enter global options for the reduction.

**To test:**

Owen had a look at the MVP pattern (now we have the MVP at three different levels) and he is happy with how the code is organized. Therefore, tester only needs to ensure that existing functionality still works. This means reading carefully the [documentation](http://docs.mantidproject.org/nightly/interfaces/ISIS_Reflectometry.html) and ensuring that the functionality is the same as on the master branch.

Please pay special attention to:
- Catalog: make sure that you can use the catalog to search for runs: in "Search Runs" select "POLREF" and enter 1520249 for the investigation id. You should be prompted with a dialog requesting login details. If you have access to the ICat a list of runs should be populated on the left. Make sure runs are transferred to the table by clicking on the "Transfer" button.
- Make sure you can Import a TBL file and reduce data: a sample TBL file can be found here [SampleData-ISIS.zip](https://github.com/mantidproject/mantid/files/361036/SampleData-ISIS.zip). You should be able to load that file into the interface ("Import TBL") and reduce the data by clicking on "Process" (if nothing is selected or if groups are selected you will need to provide a value for "Stitch1DMany" in the "Settings" tab, you can just enter "Params="-0.04"" and the reduction should work.
- Make sure you can export your table to TBL format.
- Ensure table editing features work (those can be accessed from the two menus or from the toolbar). 

Fixes #16869.
Also Fixes #16416.

Release notes were updated in this commit: https://github.com/mantidproject/mantid/pull/16941/commits/f8ce4cc0d6c481a20a6a4c4b75c577a5ab1bb199

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
